apps/choleskey/matrixutil.hpp

```diff
@@ -1,41 +1,163 @@
 #pragma once
 
-#include <iostream>
-#include <vector>
+#include <cblas.h>
+#include <lapacke.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <cstdlib>
+#include "commons.hpp"
+
+using data_type = double;
 
 // generate positive definition matrix
 template <typename T>
 using Matrix = std::vector<std::vector<T>>;
 
 template <typename T>
 std::vector<T> generate_pascal_matrix(const int n) {
-  Matrix<T> matrix(n, std::vector<T>(n, static_cast<T>(0)));
-
-  for (int i = 0; i < n; ++i) {
-    for (int j = 0; j < n; ++j) {
-      if (i == 0 || j == 0) {
-        matrix[i][j] = static_cast<T>(1);
-      } else {
-        matrix[i][j] = matrix[i][j - 1] + matrix[i - 1][j];
-      }
-    }
-  }
-
-  std::vector<T> flattenedVector;
-  for (const auto& row : matrix) {
-    flattenedVector.insert(flattenedVector.end(), row.begin(), row.end());
-  }
-  return std::move(flattenedVector);
+    Matrix<T> matrix(n, std::vector<T>(n, static_cast<T>(0)));
+
+    for (int i = 0; i < n; ++i) {
+        for (int j = 0; j < n; ++j) {
+            if (i == 0 || j == 0) {
+                matrix[i][j] = static_cast<T>(1);
+            } else {
+                matrix[i][j] = matrix[i][j - 1] + matrix[i - 1][j];
+            }
+        }
+    }
+
+    std::vector<T> flattenedVector;
+    for (const auto& row : matrix) {
+        flattenedVector.insert(flattenedVector.end(), row.begin(), row.end());
+    }
+    return std::move(flattenedVector);
 }
 
 // parameters define
 struct args_params_t : public argparse::Args {
-  bool& results = kwarg("results", "print generated results (default: false)")
-                      .set_default(true);
-  std::uint64_t& nd =
-      kwarg("nd", "Number of input(positive definition) matrix dimension(<=18)")
-          .set_default(10);
-  std::uint64_t& np = kwarg("np", "Number of partitions").set_default(4);
-  bool& help = flag("h, help", "print help");
-  bool& time = kwarg("t, time", "print time").set_default(true);
+    std::uint64_t& mat_size = kwarg("mat_size", "size of input matrix_size").set_default(10);
+    std::uint64_t& num_tiles = kwarg("num_tiles", "number of tiles").set_default(2);
+    int& verifycorrectness =
```

Comment:
Seems like a boolean so a flag like the `help` or `time` on L#44-45 should be used. Something like this:
 
```
bool& verifycorrectness = flag("verifycorrectness", "verify the tiled_cholesky results with LAPACKE_dpotrf cholesky");
```